service_container/factories.rst

```diff
@@ -38,15 +38,63 @@ configure the service container to use the
         services:
             app.newsletter_manager:
                 class:   AppBundle\Email\NewsletterManager
-                # call a static method
-                factory: ['AppBundle\Email\NewsletterManager', create]
+                # call the static method
+                factory: ['AppBundle\Email\NewsletterManagerStaticFactory', createNewsletterManager]
 
+    .. code-block:: xml
+
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <service id="app.newsletter_manager" class="AppBundle\Email\NewsletterManager">
+                    <!-- call the static method -->
+                    <factory class="AppBundle\Email\NewsletterManagerStaticFactory" method="createNewsletterManager" />
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        use AppBundle\Email\NewsletterManager;
+        use AppBundle\Email\NewsletterManagerStaticFactory;
+        use Symfony\Component\DependencyInjection\Definition;
+        // ...
+
+        $definition = new Definition(NewsletterManager::class);
+        // call the static method
+        $definition->setFactory(array(NewsletterManagerStaticFactory::class, 'createNewsletterManager'));
+
+        $container->setDefinition('app.newsletter_manager', $definition);
+
+.. note::
+
+    When using a factory to create services, the value chosen for the ``class``
+    option has no effect on the resulting service. The actual class name
+    only depends on the object that is returned by the factory. However,
```

Comment:
(This note was already there, don't know why it is marked as new by the diff.)

I do however believe the note is still valid, because it might be handy to use the class definition somewhere in a compiler pass.

On the other hand, it might be good to check why no active validation is done on the return type of the creation method...